### PR TITLE
Unpublished parts should not be viewable

### PR DESCRIPTION
**Pull request by Coenego**
We can now choose unpublished parts in the picker. However, the modules list shouldn't be shown but the view that explains to the user that this hasn't been published yet (also the view that contains the external link if enabled) should be shown.

### CODE REVIEW

**Comment by Coenego** [comment]
Assigning to @bp323 for review

File: apps/timetable/ui/js/index.js [hunk 1]
```diff
@@ -81,6 +93,15 @@ define(['gh.core', 'gh.constants', 'validator', 'gh.calendar', 'gh.header', 'gh.
      * @private
      */
     var addBinding = function() {
+
+        // Show the empty timetable page
+        $(document).on('gh.empty.timetable', function(evt, data) {
+
+            // Show the empty timetable
+            showEmptyTimetable(data);
```
Comment (bp323) [inline apps/timetable/ui/js/index.js]: Let's rewrite this as `$(document).on('gh.empty.timetable', showEmptyTimetable);` and document the event parameter on the function

**Comment by bp323** [comment]
A couple of functional issues:

- When I select a drafted part in the pickers, the view flickers from calendar to empty view to calendar again.
- When I refresh that state I get the empty view as expected, but the minimised class is not applied as expected.

**Comment by bp323** [comment]
Back to @Coenego for followup

**Comment by bp323** [comment]
Looks good, only need to sort out the module bar that shouldn't be shown.

**Comment by Coenego** [comment]
Assigning to @bp323 for review

File: apps/timetable/ui/js/index.js [hunk 2]
```diff
@@ -70,6 +59,31 @@ define(['gh.core', 'gh.constants', 'validator', 'gh.calendar', 'gh.header', 'gh.
...
 
+    /**
+     * Show the empty timetable notification
+     *
+     * @param  {Object}    evt     The dispatched jQuery event
+     * @param  {Object}    data    Object containing the module data
+     * @private
+     */
+    var showEmptyTimetable = function(evt, data) {
+        // Render the 'empty-timetable' template
```
Comment (bp323) [inline apps/timetable/ui/js/index.js]: There are 2 different formats coming in for data, you can verify this by adding an external link to a part and executing this function by: adjusting the part pickers to select the part with an external link or refreshing the page. I got this fixed by adding `data = data.record || data;` to the beginning of the file

**Comment by bp323** [comment]
Verified and merged.

**Pull request merged**